src/Renderer/LayeredMaterial.js

```diff
@@ -41,132 +43,140 @@ var getColorAtIdUv = function(nbTex) {
     return fooTexture;
 };
 
+// Array not suported in IE
+var fillArray = function(array,remp){
+
+    for(var i=0;i<array.length;i++)
+        array[i] = remp;
+};
+
+var moveElementArray = function(array,old_index, new_index)
+{
+    array.splice(new_index, 0, array.splice(old_index, 1)[0]);
+};
+
+var moveElementsArray = function (array,index, howMany, toIndex) {
+    if((toIndex > index) && (toIndex <= index + howMany)) {
+        toIndex = index + howMany;
+    }
+
+    array.splice.apply(array, [toIndex, 0].concat(array.splice(index, howMany)));
+};
+
+/* eslint-disable */
+var moveElementsArraySafe = function (array,index, howMany, toIndex) {
+    index = parseInt(index) || 0;
+    index = index < 0 ? array.length + index : index;
+    toIndex = parseInt(toIndex) || 0;
+    toIndex = toIndex < 0 ? array.length + toIndex : toIndex;
+    if((toIndex > index) && (toIndex <= index + howMany)) {
+        toIndex = index + howMany;
+    }
+
+    var moved;
+    array.splice.apply(array, [toIndex, 0].concat(moved = array.splice(index, howMany)));
+    return moved;
+};
+/* eslint-enable */
+
 var LayeredMaterial = function(id) {
 
     BasicMaterial.call(this);
 
     var maxTexturesUnits =  gfxEngine().glParams.maxTexturesUnits;
-    this.vertexShader = GlobeVS;
     var nbSamplers = Math.min(maxTexturesUnits-1,16-1);
+    this.vertexShader = GlobeVS;
 
     this.fragmentShaderHeader +='const int   TEX_UNITS   = ' + nbSamplers.toString() + ';\n';
     this.fragmentShaderHeader += pitUV;
 
     if(showDebug)
         this.fragmentShaderHeader += '#define DEBUG\n';
 
+    // see GLOBE FS
     this.fragmentShaderHeader += getColorAtIdUv(nbSamplers);
 
     this.fragmentShader = this.fragmentShaderHeader + GlobeFS;
     this.vertexShader = this.vertexShaderHeader + GlobeVS;
 
+    // handle on textures uniforms
     this.Textures = [];
-    this.pitScale = [];
-    this.nbTextures = [];
-
-    // Array not suported in IE
-    var fill = function(array,remp){
-
-        for(var i=0;i<array.length;i++)
-            array[i] = remp;
-    };
+    // handle on textures offsetScale uniforms
+    this.offsetScale = [];
+    // handle on textures number uniforms
+    this.texturesCount = [0,0];
 
     // Uniform three js needs no empty array
-    for (var l = 0; l < nbLayer; l++) {
-
-        // WARNING TODO prevent empty slot, but it's not the solution
-        this.pitScale[l] = Array(nbSamplers);
-        fill(this.pitScale[l],vector) ;
-        this.nbTextures[l] = 0;
-    }
+    // WARNING TODO: prevent empty slot, but it's not the solution
+    this.offsetScale[l_COLOR] = Array(nbSamplers);
+    this.offsetScale[l_ELEVATION] = [vector];
+    fillArray(this.offsetScale[l_COLOR],vector) ;
 
     this.Textures[0] = [emptyTexture];
     this.Textures[1] = Array(nbSamplers);
-    this.paramLayers = Array(8);
-    this.paramBLayers = Array(8);
+    var paramLayers = Array(8);
+    this.layerTexturesCount = Array(8);
```

Comment:
This is `colorLayerTexturesCount`.
Also it's a bit annoying to duplicate texture count in `layerTextureCount` and `texturesCount` :(